x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/search/aggregations/bucket/geogrid/GeoHexGridTiler.java

```diff
@@ -249,20 +249,22 @@ private static class BoundedGeoHexGridTiler extends GeoHexGridTiler {
             this.resolution = resolution;
             inflatedBboxes = new GeoBoundingBox[resolution];
             for (int i = 0; i < resolution; i++) {
-                inflatedBboxes[i] = inflateBbox(i, bbox);
+                inflatedBboxes[i] = inflateBbox(i, bbox, FACTOR);
             }
         }
 
-        private static GeoBoundingBox inflateBbox(int precision, GeoBoundingBox bbox) {
-            /*
-             * Here is the tricky part of this approach. We need to be able to filter cells at higher precisions
-             * because they are not in bounds, but we need to make sure we don't filter too much.
-             *
-             * We use h3 bins at the given resolution to check the height ands width at that level, and we add half of it.
-             *
-             * The values have been tune using test GeoHexTilerTests#testLargeShapeWithBounds
-             */
-            final double factor = FACTOR;
+        /**
+         * Since H3 cells do not fully contain their child cells, we need to take care that when
+         * filtering cells at a lower precision than the final precision, we must not exclude
+         * parents that do not match the filter, but their own children or descendents might match.
+         * For this reason the filter needs to be expanded to cover all descendent cells.
+         *
+         * This is done by taking the H3 cells at two corners, and expanding the filter width
+         * by 50% of the max width of those cells, and filter height by 50% of the max height of those cells.
+         *
+         * The inflation factor of 50% has been verified using test GeoHexTilerTests#testLargeShapeWithBounds
```

Comment:
This comment is not correct any longer?

Reply:
If you are talking about the last line, I changed that from `tuned` to `verified`, as I thought that was correct. But would you prefer we remove the line entirely?

Reply:
I was thinking more that the inflation factor is not 50% any more but 35%?

Reply:
Ah, of course. I fixed that now.